smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/endpointsV2/ParameterGenerator.java

```diff
@@ -63,6 +62,36 @@ public boolean isBuiltIn() {
         return param.expectObjectNode().containsMember("builtIn");
     }
 
+    public boolean hasDefault() {
+        return param.expectObjectNode().containsMember("default");
+    }
+
+    public String defaultAsCodeString() {
+        if (!hasDefault()) {
+            return "";
+        }
+        String buffer = "";
+        buffer += parameterName;
+        buffer += ": ";
+        buffer += "options." + parameterName + " ?? ";
+        ObjectNode paramNode = param.expectObjectNode();
+        StringNode type = paramNode.expectStringMember("type");
+
+        switch (type.getValue()) {
+            case "String":
+                buffer += "\"" + paramNode.expectStringMember("default").getValue() + "\"";
+                break;
+            case "Boolean":
+                buffer += paramNode.expectBooleanMember("default").getValue() ? "true" : "false";
+                break;
+            default:
+                // required by linter
+        }
+
+        buffer += ",";
+        return buffer;
```

Comment:
nit: Use StringBuilder?

```suggestion
        StringBuilder defaultAsCode = new StringBuilder().append(parameterName)
            .append(": options.").append(parameterName).append(" ?? ");
        ObjectNode paramNode = param.expectObjectNode();
        StringNode type = paramNode.expectStringMember("type");

        switch (type.getValue()) {
            case "String":
                defaultAsCode
                    .append("\"")
                    .append(paramNode.expectStringMember("default").getValue())
                    .append("\"");
                break;
            case "Boolean":
                defaultAsCode
                    .append(paramNode.expectBooleanMember("default").getValue() ? "true" : "false");
                break;
            default:
                // required by linter
        }

        defaultAsCode.append(",");
        return defaultAsCode.toString();
```